node crash

## OS Version: 
OSX 

## Commit hash : (if develop)
7045fc60aa588d47757de3076b59415f7d6ad342
 
## Expected behavior
Tell us what should happen.

## Actual behavior
Tell us what happens instead. Provide a log message if relevant.

## Steps to reproduce the behavior

```
panic: sync/atomic: store of inconsistently typed value into Value

goroutine 86791 [running]:
sync/atomic.(*Value).Store(0xc423496dd0, 0x47407a0, 0xc4200544d0)
	/usr/local/Cellar/go/1.10.3/libexec/src/sync/atomic/value.go:77 +0x10d
github.com/vechain/thor/comm.(*Communicator).download.func2()
	/Users/kfeng/go/src/github.com/vechain/thor/.build/src/github.com/vechain/thor/comm/sync.go:45 +0x50e
github.com/vechain/thor/co.(*Goes).Go.func1(0xc424b78f80, 0xc424128b00)
	/Users/kfeng/go/src/github.com/vechain/thor/.build/src/github.com/vechain/thor/co/goes.go:22 +0x4f
created by github.com/vechain/thor/co.(*Goes).Go
	/Users/kfeng/go/src/github.com/vechain/thor/.build/src/github.com/vechain/thor/co/goes.go:20 +0x62
```

--- a/comm/sync.go
+++ b/comm/sync.go
@@ -8,7 +8,6 @@ package comm
 import (
 	"context"
 	"fmt"
-	"sync/atomic"
 
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/pkg/errors"
@@ -26,15 +25,18 @@ func (c *Communicator) sync(peer *Peer, headNum uint32, handler HandleBlockStrea
 }
 
 func (c *Communicator) download(peer *Peer, fromNum uint32, handler HandleBlockStream) error {
+
+	// it's important to set cap to 2
+	errCh := make(chan error, 2)
+
 	ctx, cancel := context.WithCancel(c.ctx)
-	var errValue atomic.Value
 	blockCh := make(chan *block.Block, 32)
 
 	var goes co.Goes
 	goes.Go(func() {
 		defer cancel()
 		if err := handler(ctx, blockCh); err != nil {
-			errValue.Store(err)
+			errCh <- err
 		}
 	})
 	goes.Go(func() {
@@ -42,7 +44,7 @@ func (c *Communicator) download(peer *Peer, fromNum uint32, handler HandleBlockS
 		for {
 			result, err := proto.GetBlocksFromNumber(ctx, peer, fromNum)
 			if err != nil {
-				errValue.Store(err)
+				errCh <- err
 				return
 			}
 			if len(result) == 0 {
@@ -52,15 +54,15 @@ func (c *Communicator) download(peer *Peer, fromNum uint32, handler HandleBlockS
 			for _, raw := range result {
 				var blk block.Block
 				if err := rlp.DecodeBytes(raw, &blk); err != nil {
-					errValue.Store(errors.Wrap(err, "invalid block"))
+					errCh <- errors.Wrap(err, "invalid block")
 					return
 				}
 				if _, err := blk.Header().Signer(); err != nil {
-					errValue.Store(errors.Wrap(err, "invalid block"))
+					errCh <- errors.Wrap(err, "invalid block")
 					return
 				}
 				if blk.Header().Number() != fromNum {
-					errValue.Store(errors.New("broken sequence"))
+					errCh <- errors.New("broken sequence")
 					return
 				}
 				peer.MarkBlock(blk.Header().ID())
@@ -75,10 +77,13 @@ func (c *Communicator) download(peer *Peer, fromNum uint32, handler HandleBlockS
 		}
 	})
 	goes.Wait()
-	if err := errValue.Load(); err != nil {
-		return err.(error)
+
+	select {
+	case err := <-errCh:
+		return err
+	default:
+		return nil
 	}
-	return nil
 }
 
 func (c *Communicator) findCommonAncestor(peer *Peer, headNum uint32) (uint32, error) {

